plaidplotter.py

```diff
@@ -0,0 +1,318 @@
+        g = -1
+        b = -1
+        f = hue * 6 - hue_i
+        p = val * (1 - satur)
+        q = val * (1 - f * satur)
+        t = val * (1 - (1 - f) * satur)
+
+        if (0 <= hue_i and hue_i < 1):
+            r = val 
+            g = t 
+            b = p 
+        elif (1 <= hue_i and hue_i < 2):
+            r = q
+            g = val 
+            b = p 
+        elif (2 <= hue_i and hue_i < 3):
+            r = p 
+            g = val 
+            b = t 
+        elif (3 <= hue_i and hue_i < 4):
+            r = p 
+            g = q 
+            b = val 
+        elif (4 <= hue_i and hue_i < 5):
+            r = t 
+            g = p 
+            b = val 
+        else:
+            r = val 
+            g = p 
+            b = q 
+
+        r = int(r * 256)
+        g = int(g * 256)
+        b = int(b * 256)
+
+        color = 'rgb(' + str(r) + ', ' + str(g) + ', ' + str(b) + ')'
+        color = '#%02x%02x%02x' % (r, g, b)
+
+        return color
+
+    def generate_plot(self, df, args, isTrain):
+        title_str = args.name
+        path_str = args.path
+        isProspector = args.prospector
+        set_style()
+
+        # prepping data
+        col_order = (list(set(df['model'])))
+        max_time = (float(max(df['time per example (seconds)'])))
+
+        exponent = np.floor(np.log10(np.abs(max_time))).astype(int)
+        base_10 = 10
+        if exponent > 0:
+            base_10 = 1
+        else:
+            for number in range(1, np.abs(exponent)):
+                base_10 = base_10 * 10
+        max_time = ((math.ceil(base_10 * max_time)) / base_10)
+
+        # creating graph
+        palette = []
+        palette_dict = {}
+        gradient_step = .99 / (len(list(set(df['batch']))))
+        num = -1
+        golden_ratio = 0.618033988749895
+        h = random.random()
+
+        for x in df['model']:
+            if x not in palette_dict:
+                palette_dict[x] = h
+                h += golden_ratio
+                h = h % 1
+        
+        for x in palette_dict:
+            num = palette_dict[x]
+            gradient = gradient_step
+            for y in list(set(df['batch'])):
+                color = self.getColor(num, gradient, 1 - gradient)
+                palette.append(color)
+                gradient = gradient + gradient_step
+
+        fig, axes = plot(df, "model", col_order, max_time)
+        labels = (list(set(df['model'])))
+        set_labels(fig, axes, labels, list(set(df['batch'])), len(labels))
+        color_bars(axes, palette, len(df['model']), len(list(set(df['batch']))))
+
+        # saving graph
+        title = ''
+        if title_str != '':
+            title = title_str + '.png'
+        else:
+            title = time.strftime("plaidbench %Y-%m-%d-%H:%M.png")
+        print("saving figure at '" + path_str + '/' + title + "'\n")
+        fig.savefig(path_str + '/' + title)
+
+        # importing golden npy files, not other golden file utilization though
```

Comment:
Might want to split this part out into a separate function and then not call it until it does something